src/omv/modules/py_image.c

```diff
@@ -358,129 +358,112 @@ STATIC mp_obj_t py_image_getiter(mp_obj_t o_in, mp_obj_iter_buf_t *iter_buf)
     return MP_OBJ_FROM_PTR(o);
 }
 
-static void py_image_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind)
+static void py_image_print(const mp_print_t *print, mp_obj_t self, mp_print_kind_t kind)
 {
-    py_image_obj_t *self = self_in;
-    switch(self->_cobj.bpp) {
-        case IMAGE_BPP_BINARY: {
-            mp_printf(print, "{\"w\":%d, \"h\":%d, \"type\"=\"binary\", \"size\":%d}",
-                      self->_cobj.w, self->_cobj.h,
-                      ((self->_cobj.w + UINT32_T_MASK) >> UINT32_T_SHIFT) * self->_cobj.h);
-            break;
-        }
-        case IMAGE_BPP_GRAYSCALE: {
-            mp_printf(print, "{\"w\":%d, \"h\":%d, \"type\"=\"grayscale\", \"size\":%d}",
-                      self->_cobj.w, self->_cobj.h,
-                      (self->_cobj.w * self->_cobj.h) * sizeof(uint8_t));
-            break;
-        }
-        case IMAGE_BPP_RGB565: {
-            mp_printf(print, "{\"w\":%d, \"h\":%d, \"type\"=\"rgb565\", \"size\":%d}",
-                      self->_cobj.w, self->_cobj.h,
-                      (self->_cobj.w * self->_cobj.h) * sizeof(uint16_t));
-            break;
-        }
-        case IMAGE_BPP_BAYER: {
-            mp_printf(print, "{\"w\":%d, \"h\":%d, \"type\"=\"bayer\", \"size\":%d}",
-                      self->_cobj.w, self->_cobj.h,
-                      (self->_cobj.w * self->_cobj.h) * sizeof(uint8_t));
-            break;
-        }
-        default: {
-            if((self->_cobj.data[0] == 0xFE) && (self->_cobj.data[self->_cobj.bpp-1] == 0xFE)) { // for ide
-                print->print_strn(print->data, (const char *) self->_cobj.data, self->_cobj.bpp);
-            } else { // not for ide
-                mp_printf(print, "{\"w\":%d, \"h\":%d, \"type\"=\"jpeg\", \"size\":%d}",
-                          self->_cobj.w, self->_cobj.h,
-                          self->_cobj.bpp);
-            }
-            break;
-        }
+    image_t *image = py_image_cobj(self);
+    if (image->pixfmt == PIXFORMAT_JPEG
+            && image->pixels[0] == 0xFE
+            && image->pixels[image->size-1] == 0xFE) {
+        // print for ide.
+        print->print_strn(print->data, (const char *) image->pixels, image->bpp);
+    } else {
+        mp_printf(print,
+                "width: %u height: %u size:%u pixfmt=\"%s\" is_color: %u is_mutable: %u is_bayer %u is_compressed: %u",
```

Comment:
Please keep json object notation above.

Reply:
E.g. add the {} and escapes.

Reply:
Also, folks might depend on the text being the same... so, I would not change the string output.

Reply:
Reverted to old print.